2003-05-16  Jakub Jelinek  <jakub@redhat.com>

	* config/ia64/unwind-ia64.c (uw_update_reg_address): Handle
	.save XX, r0.

--- gcc-3.3.1/gcc/config/ia64/unwind-ia64.c.ia64-unwind	2003-07-07 07:19:45.000000000 +0200
+++ gcc-3.3.1/gcc/config/ia64/unwind-ia64.c	2003-07-16 11:51:02.000000000 +0200
@@ -1786,6 +1786,11 @@ uw_update_reg_address (struct _Unwind_Co
 	addr = ia64_rse_skip_regs ((unsigned long *) context->bsp, rval - 32);
       else if (rval >= 2)
 	addr = context->ireg[rval - 2].loc;
+      else if (rval == 0)
+	{
+	  static const unsigned long dummy;
+	  addr = (void *) &dummy;
+	}
       else
 	abort ();
       break;
@@ -1837,6 +1842,11 @@ uw_update_reg_address (struct _Unwind_Co
 	    context->ireg[regno - UNW_REG_R2].nat
 	      = context->ireg[rval - 2].nat;
 	  }
+	else if (rval == 0)
+	  {
+	    context->ireg[regno - UNW_REG_R2].nat.type = UNW_NAT_NONE;
+	    context->ireg[regno - UNW_REG_R2].nat.off = 0;
+	  }
 	else
 	  abort ();
 	break;
